src/scripts/Build-ModuleDoc.ps1

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+    .SYNOPSIS
+    Builds a README file in markdown for a specified module. Includes sytnax, description, required modules, parameters and examples.
+
+    .PARAMETER Path
+    Path to a .psm1 module file.
+
+    .PARAMETER Name
+    Name of a PowerShell module. (Must be discoverable by Get-Module -ListAvailable)
+
+    .PARAMETER OutputPath
+    Path to folder for resulting .md file. Defaults to $pwd.
+
+    .PARAMETER OutputFileName
+    Name for the resulting .md file. Defaults to README.md
+
+    .EXAMPLE
+    Build-ModuleDoc -Name Az
+
+    This will create a README.md for the Az module in the current working directory.
+
+    .EXAMPLE
+    Build-ModuleDoc -Path 'C:\Program Files\WindowsPowerShell\Modules\Foo\Foo.psm1'
+
+    This will create a README.md for the Foo.psm1 in the current working directory.
+
+    .EXAMPLE
+    Build-ModuleDoc -Name Foo -OutputPath 'C:\temp' -OutputFileName Bar.md
+    
+    This will create a Bar.md for the module Foo in the C:\temp directory.
+#>
+[CmdletBinding()]
+param
+(
+    [Parameter(Mandatory = $true, ParameterSetName = 'ByPath')]
+    [string]
+    [ValidateScript({Test-Path -Path $_})]
+    [ValidateScript({[System.IO.Path]::GetExtension($_) -eq '.psm1'})]
+    $Path,
+
+    [Parameter( Mandatory = $true, ParameterSetName = 'ByName')]
+    [string]
+    $Name,
+
+    [Parameter(ParameterSetName = 'ByName')]
+    [Parameter(ParameterSetName = 'ByPath')]
+    [ValidateScript({Test-Path -Path $_ -IsValid})]
+    [string]
+    $OutputPath = $pwd,
+
+    [Parameter(ParameterSetName = 'ByName')]
+    [Parameter(ParameterSetName = 'ByPath')]
+    [ValidateScript({[System.IO.Path]::GetFileNameExtension($_) -eq '.md'})]
+    [string]
+    $OutputFileName = 'README.md'
+)
+
+<#
+    .SYNOPSIS
+    Parses Comment-based help for a function and returns pertinent information.
+
+    .PARAMETER Function
+    Name of the function to parse.
+
+    .EXAMPLE
+    Get-HelpContent -Function Get-Foo
+
+    Returns an object representing the Get-Foo help content.
+#>
+function Get-HelpContent
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Function
+    )
+
+    $help = Get-Help -Name $Function -Full
+    $excludeCommonParameters = [System.Management.Automation.PSCmdlet]::CommonParameters + [System.Management.Automation.PSCmdlet]::OptionalCommonParameters
+    $examples = @()
+    $parameters = @()
+    foreach ($parameter in $help.parameters.parameter.Where({$excludeCommonParameters -notcontains $_.Name}))
```

Comment:
nit: Shouldn't these be Pascal-cased? Same for the properties on lines 90-4, 101-3, etc.
```suggestion
    foreach ($parameter in $help.parameters.parameter.Where({$excludeCommonParameters -NotContains $_.Name}))
```